src/tests/ftest/util/dfuse_utils.py

```diff
@@ -126,32 +128,64 @@ def remove_mount_point(self):
         """Remove dfuse directory
         Raises:
             CommandFailure: In case of error deleting directory
+
+        Try once with a simple rmdir which should succeed, if this
+        does not then try again with rm -rf, but still raise an error
         """
         # raise exception if mount point not specified
         if self.mount_dir.value is None:
             raise CommandFailure("Mount point not specified, "
                                  "check test yaml file")
 
-        dir_exists, _ = general_utils.check_file_exists(
+        dir_exists, clean_nodes = general_utils.check_file_exists(
             self.hosts, self.mount_dir.value, directory=True)
         if dir_exists:
+
+            target_nodes = self.hosts
+            self.log.debug('remove: hosts %s', self.hosts)
+            self.log.debug('remove: clean %s', clean_nodes)
+            if clean_nodes:
+                target_nodes.remove(clean_nodes)
+            self.log.debug('remove: target %s', target_nodes)
+
+            cmd = "rmdir {}".format(self.mount_dir.value)
+            ret_code = general_utils.pcmd(target_nodes, cmd, timeout=30)
+            if len(ret_code) == 1 and 0 in ret_code:
+                return
+
+            failed_nodes = NodeSet(",".join(
+                [str(node_set) for code, node_set in ret_code.items()
+                 if code != 0]))
+
             cmd = "rm -rf {}".format(self.mount_dir.value)
-            ret_code = general_utils.pcmd(self.hosts, cmd, timeout=30)
-            if 0 not in ret_code:
+            ret_code = general_utils.pcmd(failed_nodes, cmd, timeout=30)
+            if len(ret_code) > 1 or 0 not in ret_code:
                 error_hosts = NodeSet(
                     ",".join(
                         [str(node_set) for code, node_set in ret_code.items()
                          if code != 0]))
                 raise CommandFailure(
-                    "Error removing the {} dfuse mount point on the following "
-                    "hosts: {}".format(self.mount_dir.value, error_hosts))
+                    "Error removing the {} dfuse mount point with rm on the "
+                    "following hosts: {}".format(self.mount_dir.value,
+                                                 error_hosts))
+            raise CommandFailure(
+                "Error removing the {} dfuse mount point with rmdir on the "
+                "following hosts: {}".format(self.mount_dir.value,
+                                             failed_nodes))
 
     def run(self):
         """ Run the dfuse command.
         Raises:
             CommandFailure: In case dfuse run command fails
         """
 
+        self.log.info('Starting dfuse at %s', self.mount_dir.value)
+
+        # Allow Dfuse instances without a logfile so that they can
+        # call get_default_env(), but do not launch dfuse itself
+        # without one, as that means logs will be missing from the test.
+        assert self.log_file is not None
```

Comment:
To avoid having to possibly handle multiple exceptions raised by this method, could we raise a CommandFailure here when `not self.log_file`?

Reply:
We could do, but I thought this was all run under standard python unit testing anyway which is built on assert?
This assert should only pop if Dfuse is launched without a log_dir, so it's a failure of the test code if this happens.  I would have put it in __init__ but there are several (in my opinion misguided) cases where the code calls Dfuse(self.hostlist_clients, self.tmp).get_default_env() where obviously a log file isn't needed.

Reply:
> We could do, but I thought this was all run under standard python unit testing anyway which is built on assert?

Yes, it is and the `assert` will get the job done.   I was just thinking more generally where in some cases we call class methods with the intent of them failing - in which case it is slightly easier to deal with when there is one exception type.

> This assert should only pop if Dfuse is launched without a log_dir, so it's a failure of the test code if this happens. I would have put it in init but there are several (in my opinion misguided) cases where the code calls Dfuse(self.hostlist_clients, self.tmp).get_default_env() where obviously a log file isn't needed.

I agree on all of these points.  Why not just make the `log_dir` a required argument - `__init__()` can assert if it is set to None.  Temporarily we could include a not-really-needed `log_dir` in the few `Dfuse(self.hostlist_clients, self.tmp).get_default_env()` calls to avoid the assert.   

These `get_default_env` methods should be replaced IMHO with a call to a ServerManager object method that can return a EnvironmentVariable object populated with the requested environment variable names assigned to values that map to the current daos server config file setting / test environment setting.  Then the various classes just need to define which environment variable names need to be defined to run their command.

Reply:
> I agree on all of these points. Why not just make the `log_dir` a required argument - `__init__()` can assert if it is set to None. Temporarily we could include a not-really-needed `log_dir` in the few `Dfuse(self.hostlist_clients, self.tmp).get_default_env()` calls to avoid the assert.

That would also work, but still isn't particularly clean because you'd just be faking up a value for cases where we shouldn't be using the object in the first place - I can't really see one solution being much better than the other.  What might be best would be to make log_dir a required positional argument, then we'd have the benefit of having pylint check it for us, I'm mindful that there are probably some test codes that don't run on PRs, but I'm also wanting to get this landed because it prevents a deadlock that is blocking CI daily currently.

> These `get_default_env` methods should be replaced IMHO with a call to a ServerManager object method that can return a EnvironmentVariable object populated with the requested environment variable names assigned to values that map to the current daos server config file setting / test environment setting. Then the various classes just need to define which environment variable names need to be defined to run their command.

Sounds good, but outside the scope of this PR.